website/content/docs/v0.10/Single Board Computers/rockpi_4.md

```diff
@@ -55,3 +55,37 @@ Retrieve the admin `kubeconfig` by running:
 ```bash
 talosctl kubeconfig
 ```
+
+## Boot Talos from an SSD Drive
+
+> Note: this is only tested on Rock PI 4c
+
+Rock PI 4 has an M2 slot which supports NVMe disks. It is possible to run Talos without any SD cards right from that SSD disk.
+
+Pre-installed SPI loader won't be able to chain Talos u-boot stored on the SSD drive because it's too outdated.
+Official docs for booting from the SSD also propose using outdated SPI flash u-boot.
+
+To make it work it is necessary to update u-boot to a more recent version.
+Armbian u-boot for Rock PI 4c proved to be working: [https://users.armbian.com/piter75/](https://users.armbian.com/piter75/).
+
+### Steps
+
+- flash any OS to the SD card (can be Armbian for example).
+- download Armbian u-boot and update SPI flash:
+
+```bash
+curl -LO https://users.armbian.com/piter75/rkspi_loader-v20.11.2-trunk-v2.img
+sudo dd if=rkspi_loader-v20.11.2-trunk-v2.img of=/dev/mtdblock0 bs=4K
+```
+
+- optionally you can also write Talos image to the SSD drive right from your Rock PI board:
```

Comment:
- Optionally, you can also write Talos image to the SSD drive right from your Rock PI board: